packages/common/src/PreventExit/store.ts

```diff
@@ -0,0 +1,55 @@
+const keys: Array<symbol> = [];
+const allPreventExitState: Record<symbol, boolean> = {};
+
+function checkNoMorePreventExit() {
+  if (keys.map((key) => allPreventExitState[key]).every((i) => i === false)) {
+    window.onbeforeunload = null;
+    return true;
+  }
+  return false;
+}
+
+function publish(id: symbol, preventExit: boolean) {
```

Comment:
Add return type

Reply:
for util functions like that we always specify the return type. We don't specify it only when :
- Declaring React components
- Returning very complex types that would be too annoying to add to the return type
- Returning complex types that we don't want to manually declare because we won't export them